File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/InputsAndResultsTestByte.scala
System.out.println("getByte")
if(getByte != null){
	throw new Exception("getByte already defined!")
}
val getByte = putByte
val getByteType = getByte.getClass()
System.out.println(getByteType)
System.out.println(getByte)
if(getByteType != classOf[java.lang.Byte])
	throw new Exception("getByte not Byte!")

System.out.println("getByteVector")
if(getByteVector != null)
	throw new Exception("getByteVector already defined!")
val getByteVector = putByteVector.asInstanceOf[Array[Byte]]
val getByteVectorType = getByteVector(0).getClass()
System.out.println(getByteVectorType)
System.out.println(getByteVector)
if(getByteVectorType != classOf[Byte])
	throw new Exception("getByteVector not Byte!")

System.out.println("getByteVectorAsList")
if(getByteVectorAsList != null)
	throw new Exception("getByteVectorAsList already defined!")
val getByteVectorAsList = putByteVectorAsList.asInstanceOf[Array[Byte]]
val getByteVectorAsListType = getByteVectorAsList(0).getClass()
System.out.println(getByteVectorAsListType)
System.out.println(getByteVectorAsList)
if(getByteVectorAsListType != classOf[Byte])
	throw new Exception("getByteVectorAsList not Byte!")

System.out.println("getByteMatrix")
if(getByteMatrix != null)
	throw new Exception("getByteMatrix already defined!")
val getByteMatrix = putByteMatrix.asInstanceOf[Array[Array[Byte]]]
val getByteMatrixType = getByteMatrix(0)(0).getClass()
System.out.println(getByteMatrixType)
System.out.println(getByteMatrix)
if(getByteMatrixType != classOf[Byte])
	throw new Exception("getByteMatrix not Byte!")

System.out.println("getByteMatrixAsList")
if(getByteMatrixAsList != null)
	throw new Exception("getByteMatrixAsList already defined!")
val getByteMatrixAsList = putByteMatrixAsList.asInstanceOf[Array[Array[Byte]]]
val getByteMatrixAsListType = getByteMatrixAsList(0)(0).getClass()
System.out.println(getByteMatrixAsListType)
System.out.println(getByteMatrixAsList)
if(getByteMatrixAsListType != classOf[Byte])
	throw new Exception("getByteMatrixAsList not Byte!")

File: invesdwin-context-parent/invesdwin-context-clojure/src/test/java/de/invesdwin/context/clojure/tests/InputsAndResultsTestByte.clj
(defmacro defined [& var]
    `(try 
        (or 
            (boolean (resolve ~@var)) 
            (not (nil? (eval ~@var)))
        ) (catch Exception e# false)
    )
)

(println "getByte")
(if (defined 'getByte)
	(throw (Exception. "getByte already defined"))
)
(def getByte putByte)
(def getByteType (class getByte))
(println getByteType)
(println getByte)
(if-not (= getByteType Byte)
	(throw (Exception. "getByte not Byte!"))
)

(println "getByteVector")
(if (defined 'getByteVector)
	(throw (Exception. "getByteVector already defined"))
)
(def getByteVector putByteVector)
(def getByteVectorType (class (get getByteVector 0)))
(println getByteVectorType)
(println getByteVector)
(if-not (= getByteVectorType Byte)
	(throw (Exception. "getByteVector not Byte!"))
)

(println "getByteVectorAsList")
(if (defined 'getByteVectorAsList)
	(throw (Exception. "getByteVectorAsList already defined"))
)
(def getByteVectorAsList putByteVectorAsList)
(def getByteVectorAsListType (class (get getByteVectorAsList 0)))
(println getByteVectorAsListType)
(println getByteVectorAsList)
(if-not (= getByteVectorAsListType Byte)
	(throw (Exception. "getByteVectorAsList not Byte!"))
)

(println "getByteMatrix")
(if (defined 'getByteMatrix)
	(throw (Exception. "getByteMatrix already defined"))
)
(def getByteMatrix putByteMatrix)
(def getByteMatrixType (class (get (get getByteMatrix 0) 0)))
(println getByteMatrixType)
(println getByteMatrix)
(if-not (= getByteMatrixType Byte)
	(throw (Exception. "getByteMatrix not Byte!"))
)

(println "getByteMatrixAsList")
(if (defined 'getByteMatrixAsList)
	(throw (Exception. "getByteMatrixAsList already defined"))
)
(def getByteMatrixAsList putByteMatrixAsList)
(def getByteMatrixAsListType (class (get (get getByteMatrixAsList 0) 0)))
(println getByteMatrixAsListType)
(println getByteMatrixAsList)
(if-not (= getByteMatrixAsListType Byte)
	(throw (Exception. "getByteMatrixAsList not Byte!"))
)
